modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/DropSchemaCommand.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.catalog.commands;
+
+import static org.apache.ignite.internal.catalog.commands.CatalogUtils.schemaOrThrow;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.ignite.internal.catalog.Catalog;
+import org.apache.ignite.internal.catalog.CatalogCommand;
+import org.apache.ignite.internal.catalog.CatalogValidationException;
+import org.apache.ignite.internal.catalog.descriptors.CatalogIndexDescriptor;
+import org.apache.ignite.internal.catalog.descriptors.CatalogSchemaDescriptor;
+import org.apache.ignite.internal.catalog.descriptors.CatalogTableDescriptor;
+import org.apache.ignite.internal.catalog.storage.DropIndexEntry;
+import org.apache.ignite.internal.catalog.storage.DropSchemaEntry;
+import org.apache.ignite.internal.catalog.storage.DropTableEntry;
+import org.apache.ignite.internal.catalog.storage.UpdateEntry;
+
+/**
+ * A command that drops a schema with specified name.
+ */
+public class DropSchemaCommand implements CatalogCommand {
+    /** Returns builder to create a command to drop schema with specified name. */
+    public static DropSchemaCommandBuilder builder() {
+        return new Builder();
+    }
+
+    private final String schemaName;
+
+    private final boolean cascade;
+
+    /**
+     * Constructor.
+     *
+     * @param schemaName Name of the schema.
+     * @param cascade Flag indicating forced deletion of a non-empty schema.
+     * @throws CatalogValidationException if any of restrictions above is violated.
+     */
+    private DropSchemaCommand(String schemaName, boolean cascade) throws CatalogValidationException {
+        this.schemaName = schemaName;
```

Comment:
let's add validation for schema name (and also tests for this). See `org.apache.ignite.internal.catalog.CatalogParamsValidationUtils#validateIdentifier` for reference

Reply:
Done, see `CatalogSchemaValidationTest.testCreateSchemaWithNullOrEmptyNameIsRejected`